agent/ebs/watcher.go

```diff
@@ -131,6 +175,66 @@ func (w *EBSWatcher) overrideDeviceName(foundVolumes map[string]string) {
 	}
 }
 
+// assumes CSI Driver Managed Daemon is running else call will timeout
+func (w *EBSWatcher) StageAll(foundVolumes map[string]string) error {
+	for volID, deviceName := range foundVolumes {
+		// get volume details from attachment
+		ebsAttachment, _ := w.agentState.GetEBSByVolumeId(volID)
+		if ebsAttachment.IsSent() {
+			log.Warnf("State change event has already been emitted for EBS volume: %v.", ebsAttachment.EBSToString())
+			return nil
+		}
+		if ebsAttachment.HasExpired() {
+			log.Warnf("EBS status expired, no longer tracking EBS volume: %v.", ebsAttachment.EBSToString())
+			return nil
+		}
+		if ebsAttachment.IsAttached() {
+			return nil
+		}
+		hostPath := ebsAttachment.GetAttachmentProperties(apiebs.SourceVolumeHostPathKey)
+		filesystemType := ebsAttachment.GetAttachmentProperties(apiebs.FileSystemTypeName)
+
+		// CSI NodeStage stub equired fields
+		stubSecrets := make(map[string]string)
+		stubVolumeContext := make(map[string]string)
+		stubMountOptions := []string{}
+		stubFsGroup, _ := strconv.ParseInt("123456", 10, 8)
+		publishContext := map[string]string{"devicePath": deviceName}
+		// call CSI NodeStage
+		timeoutCtx, cancelFunc := context.WithTimeout(w.ctx, nodeStageTimeout)
+		defer func() {
+			log.Infof("cancelFunc called in node stage timeout")
+			cancelFunc()
+		}()
+		select {
+		case <-timeoutCtx.Done():
+			if timeoutCtx.Err() != nil {
+				log.Warnf("Context canceled %vs", timeoutCtx.Err())
+			}
```

Comment:
Why do we need this branch? I don't see why `timeoutCtx` would ever get cancelled? Won't we be running `default` always?